tests/unittests/Foundation/NSFileManagerTests.mm

```diff
@@ -136,4 +170,76 @@
     NSString* currentPath = [manager currentDirectoryPath];
     ASSERT_OBJCNE_MSG(originalPath, currentPath, "Expected change in current directory");
     ASSERT_OBJCEQ_MSG(parentPath, currentPath, "Expected current directory to change to parentPath");
-}
\ No newline at end of file
+}
+
+TEST(NSFileManager, MoveFileViaPath) {
+    char const* srcName = "\\NSFileManagerMoveTestFile.txt";
+    char const* destName = "\\MovedFile.txt";
+
+    NSString* content = @"The Quick Brown Fox.";
+    createFileWithContentAndVerify(srcName, content);
+
+    NSString* srcPath = getPathToFile(srcName);
+    NSString* destPath = getPathToFile(destName);
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+
+    NSError* error = nil;
+    BOOL status = [manager moveItemAtPath:srcPath toPath:destPath error:&error];
+    EXPECT_TRUE(status);
+    EXPECT_EQ(nil, error);
+
+    // Verify file exists.
+    EXPECT_TRUE([manager fileExistsAtPath:destPath]);
+    EXPECT_FALSE([manager fileExistsAtPath:srcPath]);
+
+    // Verify data.
+    NSFileHandle* fh = [NSFileHandle fileHandleForReadingAtPath:destPath];
+    EXPECT_TRUE(fh != nil);
+
+    NSData* fileData = [fh readDataToEndOfFile];
+    EXPECT_TRUE_MSG(fileData != nil, "FAILED: reading the entire file failed.");
+
+    NSString* str = [[[NSString alloc] initWithData:fileData encoding:NSUTF8StringEncoding] autorelease];
+    EXPECT_OBJCEQ_MSG(str, content, "FAILED: Unable to read the file content.");
+
+    [fh closeFile];
+
+    deleteFile((status ? destName : srcName));
```

Comment:
The deletes here worry me a bit. In a really bad situation, we could have files lying around. It might be best to move any deletes to a separate test so they get sweeped regularly instead of at the end of their own tests. Something like a test at the end of this file that checks if these files exist and deletes them if they do would be great.